storage/innobase/include/buf0buf.h

```diff
@@ -1974,7 +1990,7 @@ inline void buf_page_t::set_state(uint32_t s)
   ut_ad(s <= REMOVE_HASH || s >= UNFIXED);
   ut_ad(s < WRITE_FIX);
   ut_ad(s <= READ_FIX || zip.fix == READ_FIX);
-  zip.fix= s;
+  set_memory_state(s);
```

Comment:
As noted in the previous comment, please do not change this, but instead adjust the few places where the state is being changed from or to `NOT_USED`.